scripts/chandra/build_parent_sample.py

```diff
@@ -0,0 +1,136 @@
+    #if not os.path.exists(os.path.dirname(output_filename)):
+    #    os.makedirs(os.path.dirname(output_filename))
+
+    # Rename columns to match the standard format
+    #catalog['object_id'] = catalog['name']
+    #catalog['observation'] = catalog['obsid']
+    
+    # Process all files
+    spectra = Table(processing_fn([catalog,chandra_data_path]))
+    catalog = Table(catalog)
+    
+    catalog['name'] = catalog['name'].astype(str)
+    catalog['obsid'] = catalog['obsid'].astype(int)
+    catalog['obi'] = catalog['obi'].astype(int)
+    spectra['name'] = spectra['name'].astype(str)
+    spectra['obsid'] = spectra['obsid'].astype(int)
+    spectra['obi'] = spectra['obi'].astype(int)
+    # Join on target id with the input catalog
+    catalog = join(catalog, spectra, keys=['name','obsid','obi'], join_type='inner')
+    
+    with h5py.File(chandra_data_path+output_filename, 'w') as hdf5_file:
+        for key in catalog.colnames:
+            # Check if the column data type is a string
+            if catalog[key].dtype.kind in ['U', 'S']:
+                # Encode Unicode string to byte string
+                encoded_strings = np.char.encode(catalog[key], 'utf-8')
+                hdf5_file.create_dataset(key, data=encoded_strings)
+            elif catalog[key].dtype == 'O':  # Column with object dtype, potentially variable-length arrays
+                dt = h5py.special_dtype(vlen=np.dtype('float64'))  # or float32 if more appropriate
+                dataset = hdf5_file.create_dataset(key, (len(catalog[key]),), dtype=dt)
+                for i, item in enumerate(catalog[key]):
+                    dataset[i] = item  # Assign the variable-length array directly
+            else:
+                # Directly save the column as a dataset for non-string types
+                hdf5_file.create_dataset(key, data=catalog[key])                
+    return 1
+
+
+def main(args):
+    # Load the catalog from the catalog file, process the spectra, and save
+    # the HDF5 file
+    
+    # Open the HDF5 file
+    with h5py.File(args.file_path+args.cat_file, 'r') as hdf5_file:
+        # Initialize an empty Astropy Table
+        cat = Table()
+    
+        # Iterate over each dataset in the HDF5 file and add it as a column
+        for key in hdf5_file.keys():
+            data = hdf5_file[key][...]
+            cat.add_column(Column(data, name=key)) 
+
+
+    # Generate HDF5 file
+    data_hdf5 = save_in_standard_format([cat,args.output_file,args.file_path])
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Creates the parent sample of X-ray spectra from Chandra catalog')
+    parser.add_argument('cat_file', type=str, default='catalog.hdf5', help='Catalog file')
+    parser.add_argument('output_file', type=str, default='parent_sample_xray.hdf5', help='Name of file')
+    parser.add_argument('file_path', type=str, default='/Users/juan/science/astropile/output_data/', help='Path to spectral files')
```

Comment:
These should be consistent with the args in `download_script.py`, i.e. the paths which are required to allign should match both in default and description.

I suggest changing the default data folder to `output_data` using a relative path instead of a full (private) path.

```suggestion
    parser.add_argument('--cat_file', type=str, default='catalog.hdf5', help='Catalog file')
    parser.add_argument('--output_file', type=str, default='parent_sample_xray.hdf5', help='Name of file')
    parser.add_argument('--file_path', type=str, default='./output_data/', help='Path to spectral files')
```